src/setfit/modeling.py

```diff
@@ -295,6 +305,8 @@ def fit(
                     # to model's device
                     features = {k: v.to(device) for k, v in features.items()}
                     labels = labels.to(device)
+                    if self.model_head.multitarget:
+                        labels = labels.float()
```

Comment:
My only gripe right now is this section. In the SetFitDataset we explicitly convert the labels to `long`, as that is required for the `CrossEntropyLoss`: 
https://github.com/huggingface/setfit/blob/4fe132a2ed1b60c37acc9888021ee692cdb03ac1/src/setfit/data.py#L280

However, if multitarget is set, we *again* cast the labels, now to floats. I think we should try to modify the `SetFitDataset` such that it returns the correctly typed label from the get-go, rather than hackishly having to cast them to float mid-training.